recipes/zstd/all/conanfile.py

```diff
@@ -43,9 +43,24 @@ def configure(self):
     def layout(self):
         cmake_layout(self, src_folder="src")
 
+    def _cmake_new_enough(self, required_version):
+        try:
+            import re
+            from io import StringIO
+            output = StringIO()
+            self.run("cmake --version", output=output)
+            m = re.search(r'cmake version (\d+\.\d+\.\d+)', output.getvalue())
+            return Version(m.group(1)) >= required_version
+        except:
+            return False
+
+    def build_requirements(self):
+        # zstd/>=1.5.4 uses `check_linker_flag` which is introduced since cmake 3.1.8.
+        if Version(self.version) >= "1.5.4" and not self._cmake_new_enough("3.18"):
+            self.tool_requires("cmake/3.25.2")
```

Comment:
Shouldn't it be reported to zstd repo because it may be unintentional? Their min CMake version is still 2.8.12, so they may not want such very high min CMake version. If they want to revert, we could patch it. It's something I've seen in vulkan-loader for example, maintainers inadvertently use a recent feature in a new release, and reverted this change because they wanted to support default CMake versions of several Linux distro.

Reply:
I've opened https://github.com/facebook/zstd/issues/3500

Reply:
@SpaceIm Thanks!

Reply:
So https://github.com/facebook/zstd/pull/3510 should be backported to 1.5.4 and cmake removed from tool_requires, so that we can keep a clean conanfile.py